src/components/CallView/CallView.vue

```diff
@@ -1,75 +1,310 @@
+				const sharedData = {
+					promoted: false,
+					videoEnabled: true,
+					screenVisible: false,
+				}
+
+				this.$set(this.sharedDatas, addedModel.attributes.peerId, sharedData)
+
+				// Not reactive, but not a problem
+				this.speakingUnwatchers[addedModel.attributes.peerId] = this.$watch(function() {
+					return addedModel.attributes.speaking
+				}, function(speaking) {
+					this._setSpeaking(addedModel.attributes.peerId, speaking)
+				})
+
+				this.speakers.push({
+					id: addedModel.attributes.peerId,
+					active: false,
+				})
+
+				// Not reactive, but not a problem
+				this.screenUnwatchers[addedModel.attributes.peerId] = this.$watch(function() {
+					return addedModel.attributes.screen
+				}, function(screen) {
+					this._setScreenAvailable(addedModel.attributes.peerId, screen)
+				})
+			})
+		},
+
+		'speakers': function() {
+			this._setPromotedParticipant()
+		},
+
+		'screenSharingActive': function() {
+			this._setPromotedParticipant()
+		},
+
+		'screens': function() {
+			this._setScreenVisible()
+		},
+
+	},
+
+	mounted() {
+		this.localMediaModel = new LocalMediaModel()
+		this.localCallParticipantModel = new LocalCallParticipantModel()
+		this.callParticipantCollection = new CallParticipantCollection()
+	},
+
+	methods: {
+
+		hasDarkBackground() {
+			return this.darkBackground
+		},
+
+		_setSpeaking(peerId, speaking) {
+			if (speaking) {
+				// Move the speaker to the first element of the list
+				const index = this.speakers.findIndex(speaker => speaker.id === peerId)
+				const speaker = this.speakers[index]
+				speaker.active = true
+				this.speakers.splice(index, 1)
+				this.speakers.unshift(speaker)
+
+				return
+			}
+
+			// Set the speaker as not speaking
+			const index = this.speakers.findIndex(speaker => speaker.id === peerId)
+			const speaker = this.speakers[index]
+			speaker.active = false
+
+			// Move the speaker after all the active speakers
+			if (index === 0) {
+				this.speakers.shift()
+
+				const firstInactiveSpeakerIndex = this.speakers.findIndex(speaker => !speaker.active)
+				if (firstInactiveSpeakerIndex === -1) {
+					this.speakers.push(speaker)
+				} else {
+					this.speakers.splice(firstInactiveSpeakerIndex, 0, speaker)
+				}
+			}
+		},
+
+		_setScreenAvailable(id, screen) {
+			if (screen) {
+				this.screens.unshift(id)
+
+				return
+			}
+
+			const index = this.screens.indexOf(id)
+			if (index !== -1) {
+				this.screens.splice(index, 1)
+			}
+		},
+
+		_setPromotedParticipant() {
+			Object.values(this.sharedDatas).forEach(sharedData => {
+				sharedData.promoted = false
+			})
+
+			if (!this.screenSharingActive && this.speakers.length) {
+				this.sharedDatas[this.speakers[0].id].promoted = true
+			}
+		},
+
+		_switchScreenToId(id) {
+			const index = this.screens.indexOf(id)
+			if (index === -1) {
+				return
+			}
+
+			this.screens.splice(index, 1)
+			this.screens.unshift(id)
+		},
+
+		_setScreenVisible() {
```

Comment:
Every function is scoped to the same singlefile component in vue, you do not need to use the _ to say it's a private function or something :)

Plain text methods name are enough